### PR TITLE
Ford: Add F-150 Lightning

### PR DESCRIPTION
**Pull request by coffee-cake-isaac**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot:
- [ ] route with stock system:

No new `docs.py` due to the vehicle still being in dashcam mode.

### CODE REVIEW

**Comment by coffee-cake-isaac** [comment]
Believe the failing check might be related to this:
https://github.com/commaai/openpilot/pull/30444#issuecomment-1811503135

**Comment by coffee-cake-isaac** [comment]
@sshane Here's the F150 Lightning PR.

**Comment by sshane** [comment]
Can you give me push access to your repository?

File: selfdrive/car/ford/interface.py [hunk 1]
```diff
@@ -94,7 +100,7 @@ def _update(self, c):
     events = self.create_common_events(ret, extra_gears=[GearShifter.manumatic])
     if not self.CS.vehicle_sensors_valid:
       events.add(car.CarEvent.EventName.vehicleSensorsInvalid)
-    if self.CS.hybrid_platform:
+    if self.CS.hybrid_platform and self.CP.carFingerprint not in CANFD_CAR:
```
Comment (sshane) [inline selfdrive/car/ford/interface.py:103]: can you add a short comment explaining this?

File: selfdrive/car/ford/interface.py [hunk 2]
```diff
@@ -94,7 +100,12 @@ def _update(self, c):
     events = self.create_common_events(ret, extra_gears=[GearShifter.manumatic])
     if not self.CS.vehicle_sensors_valid:
       events.add(car.CarEvent.EventName.vehicleSensorsInvalid)
-    if self.CS.hybrid_platform:
+
+    # Ford Q3 Hybrids have an issue where sometimes an incorrect checksum is used and causes the PSCM to fault.
+    # This causes OpenPilot to have a controls mismatch and tells the user to take control immediately.
+    # This has been tested and only affects Q3 vehicles, not Q4. CAN-FD vehicles are free from this issue
+
```
Comment (adeebshihadeh) [inline selfdrive/car/ford/interface.py:107]:
```suggestion
    # Ford Q3 Hybrids have an issue where sometimes an incorrect checksum is used and causes the PSCM to fault.
```

Comment (adeebshihadeh) [inline selfdrive/car/ford/interface.py:107]:
```suggestion
    # Ford Q3 Hybrids are disabled due to occasional incorrect checksums some stock messages. This must be root caused before enabling support.
```

**Review by adeebshihadeh — APPROVED** [review]
Good to merge once the conflicts are resolved.

File: selfdrive/car/ford/interface.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/ford/interface.py:105]: we already have this in carstate.py, I meant that hybrid CANFD cars are not affected

Comment (sshane) [reply selfdrive/car/ford/interface.py:105]: https://github.com/commaai/openpilot/pull/30709

**Pull request merged**